tests/test_SALSubsystems.py

```diff
@@ -0,0 +1,152 @@
+		The file being tested
+	subsystems: `test_utils.subsystems`
+		The list of expected CSCs.
+	"""
+	# Check for known issues.
+	skip_if_known_issue("count", "none")
+	# Test SALGenerics.xml contains the expected commands.
+	root = get_file_root_element()
+	assert len(root.findall("./Subsystem/Name")) == len(ts_xml.subsystems), \
+		"There is an unexpected number of CSCs."
+
+
+def test_salsubsystems_uniq_cscs():
+	"""Test that SALSubsystems.xml does not contain duplicate CSCs.
+
+	Parameters
+	----------
+	sal_subsystems_file: `pathlib.Path(sal_interfaces/SALSubsystems.xml)`
+		The file being tested
+	subsystems: `test_utils.subsystems`
+		The list of expected CSCs.
+	"""
+	# Check for known issues.
+	skip_if_known_issue("uniq", "none")
+	# Test SALGenerics.xml contains unique CSCs.
+	root = get_file_root_element()
+	assert len(root.findall("./Subsystem/Name")) == len(set(root.findall("./Subsystem/Name"))) \
+		and len(ts_xml.subsystems) == len(set(ts_xml.subsystems)), \
+		"SALSubsystems.xml or test_utils.subsystems contains duplicate entries"
+
+
+def test_each_csc_defined():
+	"""Test that SALSubsystems.xml defines the expected set of CSCs.
+
+	Parameters
+	----------
+	sal_subsystems_file: `pathlib.Path(sal_interfaces/SALSubsystems.xml)`
+		Full filepath to SALSubsystems.xml dictionary file.
+	csc : `test_utils.subsystems`
+		Name of the CSC
+	"""
+	# Check for known issues.
+	skip_if_known_issue("defined", "none")
+	# Verify each CSC is explicitly defined.
+	root = get_file_root_element()
+	subsystems = ts_xml.subsystems
+	subsystems.sort()
+	cscs = []
+	for csc in root.findall(f"./Subsystem/Name"):
+		cscs.append(csc.text)
+	cscs.sort()
+	assert ts_xml.subsystems == cscs, "There is a duplicate CSC."
+
+
+@pytest.mark.parametrize("root,csc,generics", get_csc_generics())
+def test_generics_tag(root, csc, generics):
+	"""Test that the <Generics> tag is correctly defined for each CSC.
+
+	Parameters
+	----------
+	sal_subsystems_file: `pathlib.Path(sal_interfaces/SALSubsystems.xml)`
+		Full filepath to SALSubsystems.xml dictionary file.
+	csc : `test_utils.subsystems`
+		Name of the CSC
+	"""
+	# Check for known issues.
+	skip_if_known_issue("generics", csc)
+	# Set the condition to no for the CSCs that do not utilize the Generic topics.
+	if csc == "Script":
+		value = "no"
+	elif csc == "LOVE":
+		value = "no"
+	else:
+		value = "yes"
+	# Verify each CSC is explicitly defined.
+	assert root.find("./Subsystem/[Name='" + csc + "']/Generics").text == value, \
+		csc + " <Generics> tag is not defined as expected."
+
+
+@pytest.mark.parametrize("root,csc,simulator", get_csc_simulator())
+def test_simulator_tag(root, csc, simulator):
+	"""Test that the <Simulator> tag is correctly defined for each CSC.
+
+	Parameters
+	----------
+	sal_subsystems_file: `pathlib.Path(sal_interfaces/SALSubsystems.xml)`
+		Full filepath to SALSubsystems.xml dictionary file.
+	csc : `test_utils.subsystems`
+		Name of the CSC
```

Comment:
This list doesn't match the actual arguments (though csc does).

Reply:
Fixed